packages/core/src/shared/utilities/download.ts

```diff
@@ -28,31 +30,46 @@ export async function downloadExportResultArchive(
 ) {
     const apiStartTime = Date.now()
     let totalDownloadBytes = 0
-    const result = await cwStreamingClient.exportResultArchive(exportResultArchiveArgs)
+    let result = undefined
+    try {
+        result = await cwStreamingClient.exportResultArchive(exportResultArchiveArgs)
+        const buffer = []
 
-    const buffer = []
-
-    if (result.body === undefined) {
-        throw new ToolkitError('Empty response from Amazon Q inline suggestions streaming service')
-    }
+        if (result.body === undefined) {
+            throw new ToolkitError('Empty response from Amazon Q inline suggestions streaming service')
+        }
 
-    for await (const chunk of result.body) {
-        if (chunk.binaryPayloadEvent) {
-            const chunkData = chunk.binaryPayloadEvent
-            if (chunkData.bytes) {
-                buffer.push(chunkData.bytes)
-                totalDownloadBytes += chunkData.bytes?.length
+        for await (const chunk of result.body) {
+            if (chunk.binaryPayloadEvent) {
+                const chunkData = chunk.binaryPayloadEvent
+                if (chunkData.bytes) {
+                    buffer.push(chunkData.bytes)
+                    totalDownloadBytes += chunkData.bytes?.length
+                }
             }
         }
+        await fsCommon.writeFile(toPath, Buffer.concat(buffer))
+    } catch (e: any) {
+        const downloadErrorMessage = (e as Error).message
+        getLogger().error(`CodeTransformation: ExportResultArchive error = ${downloadErrorMessage}`)
+        telemetry.codeTransform_logApiError.emit({
+            codeTransformApiNames: 'ExportResultArchive',
+            codeTransformSessionId: CodeTransformTelemetryState.instance.getSessionId(),
+            codeTransformJobId: transformByQState.getJobId(),
```

Comment:
This gets passed in as part of `exportResultArchiveArgs` (we set `exportId` to `jobId`) -- can we use this? Anywhere we can try to move away from globals would be great, especially as we want to support simultaneous job submissions in the future

Reply:
Yes, done.